envoy/config/bootstrap/v2/bootstrap.proto

```diff
@@ -58,6 +58,9 @@ message Bootstrap {
     // configuration source.
     envoy.api.v2.core.ConfigSource cds_config = 2;
 
+    // A CDS config used for the ClusterManager's Lazy loading functionality.
+    envoy.api.v2.core.ConfigSource lazy_load_cds_config = 5;
```

Comment:
This is probably a good place to start thinking about the long term vision for CDS and how lazy loading can be supported. It's a good starting point having a separate service with the same wire format, ideally we could also repoint to lazy load from the original CDS server. 

Question: does anyone have a need for multiple CDS servers or should we only have a single CDS config source?

For the same server to handle full and incremental CDS to work, we need to have the CDS server know that a particular gRPC stream is intended to be used for lazy loading. We could basically force CDS to always be empty set of resource hints for a full dump, and then make the non-empty set the full list of resources to load lazily.

I don't think this will introduce a later barrier to incremental CDS, but want to solicit thoughts here.

Reply:
Also need to think how this works with ADS, see comment in envoyproxy/envoy#2740.

Reply:
I do not think we need to provide a way to have distinct CDS configs for lazy loading and cds. I could instead provide a bool attribute in dymanic config:

```
// Enables cluster lazy loading. Requires a valid dynamic configuration for CDS
bool enable_lazy_loading = 5;
```

Alternatively, if xDS supported incremental updates and subscriptions we would not need to maintain any lazy loading specific config.